optuna/study.py

```diff
@@ -214,7 +212,8 @@ def optimize(
             timeout=None,  # type: Optional[float]
             n_jobs=1,  # type: int
             catch=(Exception, ),  # type: Union[Tuple[()], Tuple[Type[Exception]]]
-            callbacks=None  # type: Optional[List[Callable[[Study, structs.FrozenTrial], None]]]
+            callbacks=None,  # type: Optional[List[Callable[[Study, structs.FrozenTrial], None]]]
+            disable_gc=False  # type: bool
```

Comment:
Though it's out of the original scope of this PR, `disable_gc=False` seems like double negative. How about changing the name like `gc_after_trial`?

Reply:
> `gc_after_trial`

Sounds good. I think it is more appropriate than `disable_gc`.
Thank you for your suggestion.

Fixed: https://github.com/pfnet/optuna/pull/632/commits/f384af5a2ad84daff2a360b36d676a6b6803329e.